gossip/client.go

```diff
@@ -60,23 +59,29 @@ func newClient(addr net.Addr) *client {
 // returns immediately.
 func (c *client) start(g *Gossip, done chan *client, context *rpc.Context, stopper *stop.Stopper) {
 	stopper.RunWorker(func() {
+		var err error
+
+		defer func() {
+			done <- c
+
+			if err != nil {
+				log.Infof("gossip client to %s: %s", c.addr, err)
+			}
+		}()
+
 		c.rpcClient = rpc.NewClient(c.addr, context)
 		select {
 		case <-c.rpcClient.Healthy():
-			// Success!
+			// Start gossipping and wait for disconnect or error.
```

Comment:
By the way, it's "gossiping", not "gossipping". I grepped the code and it seems there are two instances of "gossipping". Could you clean them all up?